src/analysis/normed_space/affine_isometry.lean

```diff
@@ -587,3 +588,27 @@ begin
   rw this,
   simp only [homeomorph.comp_is_open_map_iff, homeomorph.comp_is_open_map_iff'],
 end
+
+local attribute [instance, nolint fails_quickly] affine_subspace.nonempty_map
+
+include V₁
+omit V
+
+/-- Restricts an affine isometry to an affine isometry equivalence between a nonempty affine
+subspace `E` and its image. -/
+noncomputable def affine_isometry.restrict_to_equiv
+  (E : affine_subspace 𝕜 P₁) [nonempty E]
+  (φ : P₁ →ᵃⁱ[𝕜] P₂) : E ≃ᵃⁱ[𝕜] E.map φ.to_affine_map :=
+begin
+  let f := φ.to_affine_map.restrict (le_refl (E.map φ.to_affine_map)),
+  have fi : function.injective f := affine_map.restrict.injective φ.injective _,
+  have fs : function.surjective f := affine_map.restrict.surjective _ rfl,
+  have fb : function.bijective f := ⟨fi, fs⟩,
```

Comment:
Can you combine `affine_map.restrict.injective` and `affine_map.restrict.surjective` into a lemma `affine_map.restrict.bijective`?

Reply:
Would it be okay to do so in this pull request, even if it will also touch `linear_algebra/affine_space/restrict.lean`?

Reply:
Yeah, no problem! It's one lemma.